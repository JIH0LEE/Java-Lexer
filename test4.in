class Student(){
    int my_age;
    string name;
    int getAge(){
        if(age<=0){
            return "error";
        }
        else{
            return "good";
        }
    }

}