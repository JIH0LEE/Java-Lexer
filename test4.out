<ID,class>
<ID,Student>
<LPAREN>
<RPAREN>
<LBRACE>
<ID,int>
<ID,my_age>
<SEMICOLON>
<ID,string>
<ID,name>
<SEMICOLON>
<ID,int>
<ID,getAge>
<LPAREN>
<RPAREN>
<LBRACE>
<IF>
<LPAREN>
<ID,age>
<COM_OP,<=>
<INTEGER,0>
<RPAREN>
<LBRACE>
<ID,return>
<STRING,"error">
<SEMICOLON>
<RBRACE>
<ID,else>
<LBRACE>
<ID,return>
<STRING,"good">
<SEMICOLON>
<RBRACE>
<RBRACE>
<RBRACE>
