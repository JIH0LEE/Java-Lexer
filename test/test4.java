int num=0;
int num1=1;
class test4{
    int numTest=0;
    String strTest="123123";
    char charTest='2';
    boolean boolTest=false;
    int funTest(int num,boolean bool,String str, char ch){
        if(true>false){
            num=0;
            if(true==true){
                num=1;
            }
            else{
                num=2;
            }
        }
        else{
            if(true<=true){
                num=4;
            }
        }
        return ch;

    }
}