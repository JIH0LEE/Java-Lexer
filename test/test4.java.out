<1,VTYPE,int>
<1,ID,num>
<1,ASSIGN,=>
<1,NUM,0>
<1,SEMI,;>
<2,VTYPE,int>
<2,ID,num1>
<2,ASSIGN,=>
<2,NUM,1>
<2,SEMI,;>
<3,CLASS,class>
<3,ID,test4>
<3,LBRACE,{>
<4,VTYPE,int>
<4,ID,numTest>
<4,ASSIGN,=>
<4,NUM,0>
<4,SEMI,;>
<5,VTYPE,String>
<5,ID,strTest>
<5,ASSIGN,=>
<5,LITERAL,"123123">
<5,SEMI,;>
<6,VTYPE,char>
<6,ID,charTest>
<6,ASSIGN,=>
<6,CHARACTER,'2'>
<6,SEMI,;>
<7,VTYPE,boolean>
<7,ID,boolTest>
<7,ASSIGN,=>
<7,BOOLSTR,false>
<7,SEMI,;>
<8,VTYPE,int>
<8,ID,funTest>
<8,LPAREN,(>
<8,VTYPE,int>
<8,ID,num>
<8,COMMA,,>
<8,VTYPE,boolean>
<8,ID,bool>
<8,COMMA,,>
<8,VTYPE,String>
<8,ID,str>
<8,COMMA,,>
<8,VTYPE,char>
<8,ID,ch>
<8,RPAREN,)>
<8,LBRACE,{>
<9,IF,if>
<9,LPAREN,(>
<9,BOOLSTR,true>
<9,COMP,>>
<9,BOOLSTR,false>
<9,RPAREN,)>
<9,LBRACE,{>
<10,ID,num>
<10,ASSIGN,=>
<10,NUM,0>
<10,SEMI,;>
<11,IF,if>
<11,LPAREN,(>
<11,BOOLSTR,true>
<11,COMP,==>
<11,BOOLSTR,true>
<11,RPAREN,)>
<11,LBRACE,{>
<12,ID,num>
<12,ASSIGN,=>
<12,NUM,1>
<12,SEMI,;>
<13,RBRACE,}>
<14,ELSE,else>
<14,LBRACE,{>
<15,ID,num>
<15,ASSIGN,=>
<15,NUM,2>
<15,SEMI,;>
<16,RBRACE,}>
<17,RBRACE,}>
<18,ELSE,else>
<18,LBRACE,{>
<19,IF,if>
<19,LPAREN,(>
<19,BOOLSTR,true>
<19,COMP,<=>
<19,BOOLSTR,true>
<19,RPAREN,)>
<19,LBRACE,{>
<20,ID,num>
<20,ASSIGN,=>
<20,NUM,4>
<20,SEMI,;>
<21,RBRACE,}>
<22,RBRACE,}>
<23,RETURN,return>
<23,ID,ch>
<23,SEMI,;>
<25,RBRACE,}>
<26,RBRACE,}>
